apps/web/src/pages/auth/LoginPage.tsx

```diff
@@ -31,9 +29,7 @@ export default function LoginPage() {
 
   useEffect(() => {
     if (token) {
-      const user = jwtDecode<IJwtPayload>(token);
-
-      if (!invitationToken && (!user.organizationId || !user.environmentId)) {
+      if (!invitationToken && currentUser?._id && (!currentUser?.organizationId || !currentUser?.environmentId)) {
```

Comment:
The additional check here for `currentUser?._id` ensures that the User is updated in Auth Context before a redirection can take place, otherwise the page will flicker to `ROUTES.AUTH_APPLICATION` when loading.